torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1991,17 +1993,50 @@ def _backward_impl(ctx, all_args):
                     context = torch._C._DisableAutocast if disable_amp else nullcontext
                     with tracing(saved_context), compile_context(
                         saved_compile_context
-                    ), context(), track_graph_compiling(aot_config, "backward"):
-                        CompiledFunction.compiled_bw = aot_config.bw_compiler(
-                            bw_module, placeholder_list
-                        )
-                        # Maybe save cache entry
-                        if try_save_cache_entry is not None:
-                            try_save_cache_entry(
-                                CompiledFunction.compiled_bw,
-                                fw_metadata,
-                                aot_config,
+                    ), context(), track_graph_compiling(
+                        aot_config, "backward"
+                    ), METRICS_CONTEXT, dynamo_timed(
+                        # TODO: Capture this under "entire_frame_compile" or a new name?
+                        # It would seem to serve the same purpose when we capture the full
+                        # compile in convert_frame_py, but "frame" may not make sense here.
+                        "entire_frame_compile",
+                        fn_name="backward._backward_impl",
+                        log_pt2_compile_event=False,
+                        # TODO: How do we want to record this one in dynamo_compile?
+                        dynamo_compile_column="dynamo_cumulative_compile_time_us",
+                    ):
+                        fail_type: Optional[str] = None
+                        fail_reason: Optional[str] = None
+                        start_ns = time.time_ns()
+                        try:
+                            CompiledFunction.compiled_bw = aot_config.bw_compiler(
+                                bw_module, placeholder_list
                             )
+                            # Maybe save cache entry
+                            if try_save_cache_entry is not None:
+                                try_save_cache_entry(
+                                    CompiledFunction.compiled_bw,
+                                    fw_metadata,
+                                    aot_config,
+                                )
+                        except Exception as e:
+                            fail_type = type(e).__qualname__
+                            fail_reason = str(e)
```

Comment:
I don't think there's a good reason to force this handling at every call site. `dynamo_timed` is 100% capable of catching exceptions and adding them to the field before continuing propagation.

Reply:
Planning to do in a follow-up diff.

Reply:
added TODO